### DISCUSSION TITLE
validation failed: Cast to string failed (type Array)

### DISCUSSION THREAD

**Discussion by arashi-dev** [typegoose/typegoose#635] [Q&A] [answered]
Hello.
I'm trying to use an array of strings but it fails.

This is the model:
```ts
class Product {
    @Prop({ type: () => String })
    category?: string[];
}

const ProductModel = getModelForClass(Product)
```

This is how I create new document:
```ts
await ProductModel.create({ category: [] } as Product)
```

And this is the error I get:
```ts
"Product validation failed: category: Cast to string failed for value \"[]\" (type Array) at path \"category\""
```

All primitive and embedded documents work but I get error with these arrays value

**Comment by arashi-dev** [comment, marked as answer]
I found the solution.
I had to use the `@Prop` like this

```ts
@Prop({ type: () => [String] }, WhatIsIt.ARRAY)
category?: string[];
```

**Reply by hasezoey** [reply]
did you have in your tsconfig:
- `target` set to `es6` (or higher)?
- `emitDecoratorMetadata` set to `true`?

For reference, read [Use Without "emitDecoratorMetadata"](https://typegoose.github.io/typegoose/docs/guides/use-without-emitDecoratorMetadata)

**Reply by arashi-dev** [reply]
thank you for your respond.
Yes. I am using `es2018` as the `target` and `emitDecoratorMetadata`
This is my config
```JSON
"compilerOptions": {
    "target": "es2018", <----
    "lib": [
      "dom",
      "dom.iterable",
      "esnext",
      "ES2018",
      "ESNext.AsyncIterable"
    ],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": false,
    "forceConsistentCasingInFileNames": true,
    "noEmit": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve",
    "incremental": true,
    "emitDecoratorMetadata": true,  <----
    "experimentalDecorators": true
  }
```

**Reply by hasezoey** [reply]
thanks, do you use any other compiler than `tsc` / `ts-node` / `ts-jest`? (without the options of [`transpile-only`](https://typegoose.github.io/typegoose/docs/guides/known-issues#transpile-only))

**Reply by arashi-dev** [reply]
yes. babel which comes from NextJS

**Reply by hasezoey** [reply]
did you read & apply [Known Issues: Babel](https://typegoose.github.io/typegoose/docs/guides/known-issues#babel) (in this case `babel-plugin-transform-typescript-metadata` for `emitDecoratorMetadata`)?
